File: sdk-cpp/examples/metadata.cpp
/// \file
/// Metadata ErrorBrain C++ SDK example

#include <errorbrain/errorbrain.h>
#include <iostream>
#include <map>

int main() {
    std::cout << "ErrorBrain C++ SDK - Metadata Example" << std::endl;
    std::cout << "====================================" << std::endl << std::endl;

    try {
        ErrorBrain::ErrorBrainClient client;

        // Create comprehensive metadata
        std::map<std::string, std::string> metadata{
            {"endpoint", "/api/v1/users"},
            {"method", "POST"},
            {"client_ip", "203.0.113.42"},
            {"requests_per_minute", "1200"},
            {"limit", "1000"},
            {"user_id", "user_12345"},
        };

        ErrorBrain::ErrorReport report("cpp", "api-gateway", "Rate limit exceeded");
        report.with_tags({"api", "rate-limit", "production"})
            .with_metadata(metadata)
            .with_store_in_vault(true);

        auto response = client.send_error(report);
        std::cout << "✅ Error with metadata sent" << std::endl;
        std::cout << "   ID: " << response.id() << std::endl;
        std::cout << "   Tags: ";
        for (size_t i = 0; i < response.tags().size(); ++i) {
            if (i > 0) std::cout << ", ";
            std::cout << response.tags()[i];
        }
        std::cout << std::endl;
        std::cout << "   Analysis: " << response.explanation() << std::endl;

    } catch (const ErrorBrain::ErrorBrainException& e) {
        std::cerr << "❌ Failed to send error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

File: sdk-rust/examples/metadata.rs
//! Rust/Cargo example: Error with rich metadata

use errorbrain_sdk::{ErrorBrainClient, ErrorReport};
use std::collections::HashMap;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("ErrorBrain Rust SDK - Metadata Example");
    println!("====================================\n");

    let client = ErrorBrainClient::default();

    // Create comprehensive metadata
    let mut metadata = HashMap::new();
    metadata.insert("endpoint".to_string(), "/api/v1/users".to_string());
    metadata.insert("method".to_string(), "POST".to_string());
    metadata.insert("client_ip".to_string(), "203.0.113.42".to_string());
    metadata.insert(
        "requests_per_minute".to_string(),
        "1200".to_string(),
    );
    metadata.insert("limit".to_string(), "1000".to_string());
    metadata.insert("user_id".to_string(), "user_12345".to_string());

    let report = ErrorReport::new("rust", "api-gateway", "Rate limit exceeded")
        .with_tags(vec![
            "api".to_string(),
            "rate-limit".to_string(),
            "production".to_string(),
        ])
        .with_metadata(metadata)
        .with_store_in_vault(true);

    match client.send_error(&report).await {
        Ok(response) => {
            println!("✅ Error with metadata sent");
            println!("   ID: {}", response.id);
            println!("   Tags: {}", response.tags.join(", "));
            println!("   Analysis: {}", response.explanation);
        }
        Err(e) => {
            eprintln!("❌ Failed to send error: {}", e);
        }
    }

    Ok(())
}
